code/mob/living/critter/ai/mimic.dm

```diff
@@ -0,0 +1,155 @@
+	var/mob/living/critter/mimic/C = holder.owner
+	return C.is_hiding && C.can_critter_attack()
+
+/datum/aiTask/sequence/goalbased/ambush/get_targets()
+	var/mob/living/critter/C = holder.owner
+	return C.seek_target(src.max_dist)
+
+/datum/aiTask/sequence/goalbased/ambush/on_tick()
+	var/mob/living/critter/mimic/C = holder.owner
+	if(C.is_hiding)
+		C.stop_hiding()
+	. = ..()
+
+/datum/aiTask/succeedable/ambush
+	name = "ambush subtask"
+	max_dist = 5
+	var/has_started = FALSE
+
+/datum/aiTask/succeedable/ambush/failed()
+	//failure condition is just that the target escaped
+	if(holder.owner && holder.target)
+		return (GET_DIST(holder.owner, holder.target) > max_dist)
+	else
+		return TRUE //we also fail if owner or target are somehow null
+
+/datum/aiTask/succeedable/ambush/succeeded()
+	if(is_incapacitated(holder.target)) //attack until target is incapacitated
+		return TRUE
+	return FALSE
+
+/datum/aiTask/succeedable/ambush/on_tick()
+	//keep moving towards the target and attacking them in range for as long as is necessary
+	//has_started marks that we've hit them once
+	var/mob/living/critter/C = holder.owner
+	var/mob/M = holder.target
+	if(C && M && BOUNDS_DIST(C, M) == 0)
+		C.set_dir(get_dir(C, M))
+		if(C.can_critter_attack()) //if we can't attack, just do nothing until we can
+			C.critter_attack(holder.target)
+			src.has_started = TRUE
+	else if(C && M)
+		//we're not in punching range, let's fix that by moving back to the move subtask
+		var/datum/aiTask/sequence/goalbased/ambush/parent_task = holder.current_task
+		parent_task.current_subtask = parent_task.subtasks[1] //index 1 is always the move task in goalbased
+		parent_task.subtask_index = 1
+		parent_task.current_subtask.reset()
+
+/datum/aiTask/succeedable/ambush/on_reset()
+	src.has_started = FALSE
+	var/mob/living/critter/C = holder.owner
+	if(C)
+		C.set_a_intent(INTENT_HARM)	//we an angry critter
+
+
+/// If a target can see us and we're not hiding, then lets run away and hide
+/datum/aiTask/sequence/goalbased/run_and_hide
+	name = "hiding"
+	weight = 11 //lower than ambush & waiting in ambush, but higher than attack
+	max_dist = 11
+	var/min_dist = 7
+
+/datum/aiTask/sequence/goalbased/run_and_hide/New(parentHolder, transTask)
+	..()
+	add_task(holder.get_instance(/datum/aiTask/succeedable/run_and_hide, list(holder)))
+
+/datum/aiTask/sequence/goalbased/run_and_hide/precondition()
+	var/mob/living/critter/mimic/C = holder.owner
+	var/datum/targetable/critter/mimic/mimic_ability = C.abilityHolder.getAbility(/datum/targetable/critter/mimic)
+	return !C.is_hiding && !mimic_ability.disabled && mimic_ability.cooldowncheck()
+
+//list of items in a range min to max that are not visible to other mobs
+/datum/aiTask/sequence/goalbased/run_and_hide/get_targets()
+	. = list()
+	for(var/turf/T in orange(holder.owner, max_dist))
+		if(GET_DIST(holder.owner, T) < min_dist)
+			continue
+		for(var/mob/living/carbon/human/H in view(T, 3))
```

Comment:
```suggestion
		for(var/mob/living/carbon/human/H in viewers(T, 3))
```
More correct for 'who can see this turf', but does mean that humans with mesons/xray will mark all the turfs they're close to as visible? Not sure if that's worth it, up to you.